### PR TITLE
Fixed Ac_mul_Bt and added export and tests, added a missing lapack test

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by tkelman** [comment]
travis failure was #15294, restarted
```
	From worker 4:	     * numbers              Error During Test
	From worker 4:	  Test threw an exception of type UndefVarError
	From worker 4:	  Expression: round(T,true // false) === convert(T,Inf)
	From worker 4:	  UndefVarError: no_op_err not defined
	From worker 4:	   in gcd(::Bool, ::Bool) at ./intfuncs.jl:11
	From worker 4:	   in Rational{Bool}(::Bool, ::Bool) at ./rational.jl:9
	From worker 4:	   [inlined code] from /tmp/julia/share/julia/test/numbers.jl:2849
	From worker 4:	   in anonymous at ./no file:4294967295
	From worker 4:	   [inlined code] from ./essentials.jl:79
	From worker 4:	   in include_string(::UTF8String, ::ASCIIString) at ./loading.jl:371
	From worker 4:	   in include_from_node1(::ASCIIString) at ./loading.jl:420
	From worker 4:	   [inlined code] from ./util.jl:179
	From worker 4:	   in runtests(::ASCIIString) at /tmp/julia/share/julia/test/testdefs.jl:7
	From worker 4:	   in (::Base.Serializer.__deserialized_types__.##9186)(::ASCIIString) at /tmp/julia/share/julia/test/runtests.jl:36
	From worker 4:	   in run_work_thunk(::Base.##254#256{Base.CallMsg{:call_fetch}}, ::Bool) at ./multi.jl:714
	From worker 4:	   [inlined code] from ./multi.jl:1010
	From worker 4:	   in (::Base.##253#255{Base.CallMsg{:call_fetch},TCPSocket})() at ./task.jl:59
```

File: test/linalg/matmul.jl [hunk 1]
```diff
@@ -164,6 +165,21 @@ let A = [1+2im 3+4im; 5+6im 7+8im], B = [2+7im 4+1im; 3+8im 6+5im]
     end
 end
 
+#vecdot for generic reals as well
+let A = [1 3; 5 7], B = [2 4; 3 6]
+    @test vecdot(A,B) == dot(vec(A),vec(B)) == vecdot_(A,B) == vecdot(float(A),float(B))
+    @test vecdot(Int[], Int[]) == 0 == vecdot_(Int[], Int[])
+    @test_throws MethodError vecdot(Any[], Any[])
```
Comment (timholy) [inline test/linalg/matmul.jl]: Why is it important to test for this?

Comment (kshyatt) [inline test/linalg/matmul.jl]: https://codecov.io/github/JuliaLang/julia/base/linalg/matmul.jl?ref=1eacbbbb9a686292a12e89bcc4d9559c22928b83#l-46

**Comment by kshyatt** [comment]
CI passed! Anyone have further thoughts?

File: test/linalg/matmul.jl (hunk 1 above)
Comment (timholy) [inline test/linalg/matmul.jl]: I'm being dense, but that line seems to require a `BlasReal` and this is `Any`. The test seems to call https://codecov.io/github/JuliaLang/julia/base/linalg/generic.jl?ref=1eacbbbb9a686292a12e89bcc4d9559c22928b83#l-270, which is (mostly) covered, and which ends up being a test of whether `zero(Any)` is defined.

Comment (kshyatt) [inline test/linalg/matmul.jl]: I think you're right. I'll remove this for now and we can come back to it later.

File: test/linalg/matmul.jl [hunk 2]
```diff
@@ -59,6 +59,7 @@ A = rand(1:20, 5, 5) .- 10
 B = rand(1:20, 5, 5) .- 10
 @test At_mul_B(A, B) == A'*B
 @test A_mul_Bt(A, B) == A*B'
+@test Ac_mul_Bt(A, B) == A'*B.'
```
Comment (tkelman) [inline test/linalg/matmul.jl]: should this actually get tested on complex inputs so we can be sure the conjugation behavior is right?

**Comment by andreasnoack** [comment]
Hm. We actually don't have parsing for `Ac_mul_Bt`
```jl
julia> expand(:(A'*B.'))
:(Ac_mul_B(A,transpose(B)))
```
which probably explains why the bug has never surfaced. We should maybe just delete the methods.

**Comment by kshyatt** [comment]
@andreasnoack do you want me to delete the methods, or should we add the parsing?

**Comment by andreasnoack** [comment]
Let's delete them.

**Pull request closed without merging**